modules/core/src/main/scala/ch/epfl/bluebrain/nexus/iam/core/User.scala

```diff
@@ -26,6 +26,13 @@ final case class AnonymousUser()(implicit prefix: IdentityIdPrefix = IdentityIdP
   override val identities = Set(Identity.Anonymous())
 }
 
+/**
+  * Singleton representing any service account, i.e. a user belonging to the Nexus internal realm.
+  */
+case object ServiceAccount extends User {
+  override val identities: Set[Identity] = Set.empty
```

Comment:
If we set this to an empty list, the events stored when the user logged is a `ServiceAccount` will be stored for the user `Anonymous`. I would suggest we create a user for it (hardcoded) or that we just allow read operations (which do not store any metadata information into the primary store).